Sources/Datadog/Core/DatadogCoreProtocol.swift

```diff
@@ -0,0 +1,54 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2019-2020 Datadog, Inc.
+ */
+
+import Foundation
+
+internal var DefaultDatadogCore: DatadogCoreProtocol = NOOPDatadogCore()
+
+/// A Datadog Feature registry hold a set of features and is responsible of
+/// managing their storage and upload mechanism. It also provide thread-safe
+/// scope for features.
+public protocol DatadogCoreProtocol {
+    /// Registers a feature by its name and configuration.
+    ///
+    /// - Parameters:
+    ///   - named: The feature name.
+    ///   - storage: The feature's storage configuration.
+    ///   - upload: The feature's upload configuration.
+    func registerFeature(named: String, storage: FeatureStorageConfiguration, upload: FeatureUploadConfiguration)
+
+    /// Returns a Feature scope by its name.
+    ///
+    /// - Parameter named: The feature's name.
+    /// - Returns: The feature if it was previously registered, `nil` otherwise.
+    func scope(forFeature named: String) -> FeatureScope?
+}
+
+/// Provide feature specific storage configuration.
+public struct FeatureStorageConfiguration {
+    // TODO: RUMM-2133
+}
+
+/// Provide feature specific upload configuration.
+public struct FeatureUploadConfiguration {
+    // TODO: RUMM-2133
+}
+
```

Comment:
As we discussed in RFC comment, let's make sure we can resign from having `SDKFeatureConfiguration` protocol:
```swift
public protocol SDKFeatureConfiguration {
    func register(to sdkCore: SDKCore)
}
```
without resigning from type safety in features configuration (generic `FeatureStorageConfiguration<Metadata>` and `FeatureUploadConfiguration<Metadata>`). From my experiments in POC, this interface appears to be required, but I might be missing something. Not a blocker for this PR, but something we should discuss 🙌.

Reply:
Yes, this strongly depends on how we will expect users to initialize features and SDK instances. We can discuss first and see how we move forward 👍